src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,311 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+/**
+ * The Parser class is responsible for parsing user input commands and executing corresponding actions.
+ * It interacts with the TaskList, Ui, and Storage classes to manage tasks and data.
+ */
+
+public class Parser {
+    private Scanner scanner;
+    private TaskList tasklist;
+    private Ui ui;
+    private Storage storage;
+
+    public Parser(Scanner s, TaskList t, Storage st) {
+        scanner = s;
+        tasklist = t;
+        ui = new Ui();
+        storage = st;
+    }
+
+    /**
+     * Reads user input commands, processes them, and executes corresponding actions.
+     * Continuously loops until the user exits the program.
+     */
+    public void read() {
+        while (scanner.hasNext()) {
+
+            String userInput = scanner.nextLine();
+            String userInputLowercase = userInput.toLowerCase();
+
+            if (checkFeeding(userInputLowercase)) {
+                ui.happy();
+                continue;
+            } else if (checkIfBaseCommand(userInputLowercase)) {
+                handleBaseCommand(userInput.split(" "));
+                storage.writeToFile(tasklist);
+            } else if (checkIfFind(userInputLowercase)) {
+                handleFind(userInput.split(" "));
+                storage.writeToFile(tasklist);
+            } else if (checkIfLeave(userInputLowercase)) {
+                ui.goodbye();
+                break;
+            } else if (checkIfList(userInputLowercase)) {
+                handleList(tasklist);
+            } else if (checkIfTodo(userInputLowercase)) {
+                handleTodo(userInput, tasklist);
+                storage.writeToFile(tasklist);
+            } else if (checkIfEvent(userInputLowercase)) {
+                handleEvent(userInput, tasklist);
+                storage.writeToFile(tasklist);
+            } else if (checkIfDeadline(userInputLowercase)) {
+                handleDeadline(userInput, tasklist);
+                storage.writeToFile(tasklist);
+            } else {
+                ui.instructionMessage();
+            }
+        }
+    }
+
+    /**
+     * Handles the user input command for adding an event task.
+     * Parses the input string, creates an event task, and adds it to the task list.
+     *
+     * @param s The user input command string for adding an event task.
+     * @param t The task list to which the event task will be added.
+     */
+    public void handleEvent(String s, TaskList t) {
```

Comment:
Refactor validation logic into separate methods to improve readability and maintainability. For example, extracting the validation of event dates into a separate method would clarify the handleEvent method.